libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts

```diff
@@ -22,20 +24,34 @@ export class GenericAdrLicenseApi implements GenericLicenseClient<AdrDto> {
     private adrApi: AdrApi,
   ) {}
 
+  handleError(error: any): any {
+    this.logger.error('ADR license fetch failed', {
+      exception: error,
+      message: (error as Error)?.message,
+      category: LOG_CATEGORY,
+    })
+
+    throw new ApolloError(
+      'Failed to resolve request',
+      error?.message ?? error?.response?.message,
+    )
+  }
+
+  private handle4xx(error: FetchError) {
```

Comment:
I'd merge this just with handleError with a comment about not logging these statuses. Or maybe log them as info? It could be relevant if e.g. a user thinks they should be seeing a license but the service returns 404 for them.